src/server/scripts/Pandaria/GateoftheSettingSun/boss_striker_gadok.cpp

```diff
@@ -0,0 +1,752 @@
+    MOV_SOUTH_NORTH         = 1,
+    MOV_WEST_EAST           = 2,
+    MOV_EAST_WEST           = 3
+};
+
+StrafPointStruct StrafPoints[4] =
+{
+    { POINT_NORTH_OUTSIDE, {1258.0f, 2304.644f, 438.0f, 0.0f}, POINT_NORTH_START, {1238.007f, 2304.644f, 435.0f, 0.0f}, POINT_NORTH_END, {1153.398f, 2304.578f, 435.0f, 0.0f}, POINT_SOUTH_OUTSIDE, {1133.4f, 2304.578f, 438.0f, 0.0f} }, // North -> South
+    { POINT_SOUTH_OUTSIDE, {1133.4f, 2304.578f, 438.0f, 0.0f}, POINT_SOUTH_START, {1153.398f, 2304.578f, 435.0f, 0.0f}, POINT_SOUTH_END, {1238.007f, 2304.644f, 435.0f, 0.0f}, POINT_NORTH_OUTSIDE, {1258.0f, 2304.644f, 438.0f, 0.0f} }, // South -> North
+    { POINT_WEST_OUTSIDE,  {1195.3f, 2366.941f, 438.0f, 0.0f}, POINT_WEST_START,  {1195.299f, 2348.941f, 435.0f, 0.0f}, POINT_WEST_END,  {1195.392f, 2263.441f, 435.0f, 0.0f}, POINT_EAST_OUTSIDE,  {1195.4f, 2243.441f, 438.0f, 0.0f} }, // West  -> East
+    { POINT_EAST_OUTSIDE,  {1195.4f, 2243.441f, 438.0f, 0.0f}, POINT_EAST_START,  {1195.392f, 2263.441f, 435.0f, 0.0f}, POINT_EAST_END,  {1195.299f, 2348.941f, 435.0f, 0.0f}, POINT_WEST_OUTSIDE,  {1195.3f, 2366.941f, 438.0f, 0.0f} }  // East  -> West
+};
+
+Position CenterPos = {1195.0f, 2304.0f, 438.0f};
+
+enum eSpells
+{
+    SPELL_PREY_TIME         = 106933,
+    SPELL_IMPALING_STRIKE   = 107047,
+
+    SPELL_STRAFING_RUN      = 107342,
+    SPELL_STRAFIND_RUN_DMG  = 116298,
+
+    SPELL_RIDE_VEHICLE      = 46598,
+
+    // Disruptor
+    SPELL_BOMB              = 115110
+};
+
+enum eEvents
+{
+    EVENT_PREY_TIME         = 1,
+    EVENT_IMPALING_STRIKE   = 2,
+
+    EVENT_DISRUPTOR_BOMBARD = 3,
+};
+
+enum ePhases
+{
+    PHASE_MAIN          = 1,
+    PHASE_NORTH_SOUTH   = 2,
+    PHASE_WEST_EAST     = 3
+};
+
+enum eStrafing
+{
+    STRAF_NONE    = 0,
+    STRAF_70      = 1,
+    STRAF_30      = 2
+};
+
+#define MAX_DISRUPTOR   5
+#define MAX_STRIKER     10
+#define RADIUS_CIRCLE   100.0f
+
+class boss_striker_gadok : public CreatureScript
+{
+    public:
+        boss_striker_gadok() : CreatureScript("boss_striker_gadok") {}
+
+        struct boss_striker_gadokAI : public BossAI
+        {
+            boss_striker_gadokAI(Creature* creature) : BossAI(creature, DATA_GADOK)
+            {
+                instance = creature->GetInstanceScript();
+            }
+
+            InstanceScript* instance;
+
+            bool isStrafing;
+
+            uint32 strafingTimer;
+
+            uint8 strafingEventCount;
+            uint8 strafingEventProgress;
+            uint8 move;
+
+            void Reset()
+            {
+                _Reset();
+                isStrafing = false;
+
+                strafingTimer = 0;
+
+                strafingEventCount = 0;
+                strafingEventProgress = 0;
+                move = 0;
+                me->GetMotionMaster()->Clear();
+
+                summons.DespawnAll();
+                DoSpawnKrikThik();
+
+                if (instance)
+                    instance->SetData(DATA_GADOK, PHASE_MAIN);

```

Comment:
sure? _Reset() changes boss state of DATA_GADOK, is that intended or some leftover?